GetMessage/handler.ts

```diff
@@ -174,169 +165,188 @@ export const getReadStatusForService = (
-
-    if (E.isLeft(errorOrMaybeMessageStatus)) {
-      return ResponseErrorInternal(
-        `Error retrieving MessageStatus: ${JSON.stringify(
-          errorOrMaybeMessageStatus.left
-        )}`
-      );
-    }
-    const maybeMessageStatus = errorOrMaybeMessageStatus.right;
-
-    const returnedMessage = pipe(
-      {
-        message,
-        notification: pipe(notificationStatuses, O.toUndefined),
-        // we do not return the status date-time
-        status: pipe(
-          maybeMessageStatus,
-          O.map(messageStatus => messageStatus.status),
-          // when the message has been received but a MessageStatus
-          // does not exist yet, the message is considered to be
-          // in the ACCEPTED state (not yet stored in the inbox)
-          O.getOrElse(() => MessageStatusValueEnum.ACCEPTED)
-        )
-      },
-      // Enrich message info with advanced properties if user is allowed to read them
-      messageWithoutAdvancedProperties =>
-        B.fold(
-          () => messageWithoutAdvancedProperties,
-          () => ({
-            ...messageWithoutAdvancedProperties,
-            read_status: B.fold(
-              () => ReadStatusEnum.UNAVAILABLE,
-              () => getReadStatusForService(maybeMessageStatus)
-            )(canReadMessageReadStatus(auth.subscriptionId))
-          })
-        )(canReadAdvancedMessageInfo(message, auth.groups))
+    )
+  )();
+
+  if (E.isLeft(errorOrMaybeContent)) {
+    context.log.error(
+      `GetMessageHandler|${JSON.stringify(errorOrMaybeContent.left)}`
     );
+    return errorOrMaybeContent.left;
+  }
+
+  const content = O.toUndefined(errorOrMaybeContent.right);
 
-    return ResponseSuccessJson(returnedMessage);
+  const message = {
+    content,
+    ...retrievedMessageToExternal(retrievedMessage)
   };
-}
+
+  const errorOrNotificationStatuses = await getMessageNotificationStatuses(
+    notificationModel,
+    notificationStatusModel,
+    retrievedMessage.id
+  )();
+
+  if (E.isLeft(errorOrNotificationStatuses)) {
+    return ResponseErrorInternal(
+      `Error retrieving NotificationStatus: ${errorOrNotificationStatuses.left.name}|${errorOrNotificationStatuses.left.message}`
+    );
+  }
+  const notificationStatuses = errorOrNotificationStatuses.right;
+
+  const errorOrMaybeMessageStatus = await messageStatusModel.findLastVersionByModelId(
+    [retrievedMessage.id]
+  )();
+
+  if (E.isLeft(errorOrMaybeMessageStatus)) {
+    return ResponseErrorInternal(
+      `Error retrieving MessageStatus: ${JSON.stringify(
+        errorOrMaybeMessageStatus.left
+      )}`
+    );
+  }
+  const maybeMessageStatus = errorOrMaybeMessageStatus.right;
+
+  const errorOrServiceCanReadMessageReadStatus = await canAccessMessageReadStatus(
+    auth.subscriptionId,
+    fiscalCode
+  )();
+
+  if (E.isLeft(errorOrServiceCanReadMessageReadStatus)) {
+    return ResponseErrorInternal(
+      `Error retrieving information about read status preferences: ${JSON.stringify(
```

Comment:
IMHO is better a more opaque message, adding instead a log error for our debug.

Reply:
Agree. Done in [5665e31](https://github.com/pagopa/io-functions-services/pull/211/commits/5665e310ad0fe72c41bc14bc37bd93db10ee80c9) thanks